src/main/scala/resource/Using.scala

```diff
@@ -0,0 +1,117 @@
+package resource
+
+import java.io._
+import java.net.URL
+import java.nio.channels.FileChannel
+import java.nio.charset.Charset
+import java.util.jar.{JarInputStream, JarOutputStream, JarFile}
+import java.util.zip._
+
+/** Convenience methods for common java IO operations.
+  *
+  *
+  * This API attempts to do two things:
+  *
+  * 1. Clean up common java IO operations and conversions, e.g. always returning "Buffered" streams rather than raw ones.
+  * 2. Avoid frustrating errors when dealing with the filesystem, e.g. trying to write to a file whose parent directories
+  *    do not exist yet.
+  *
+  * Note: This code is ported from the sbt IO library.
+  */
+object Using {
```

Comment:
added a few.  hard to actually check, since the handle to the file is transient in the closure, without capturing it via evil mutable state.